config/config.go

```diff
@@ -570,12 +561,9 @@ func (c *AlertmanagerConfig) UnmarshalYAML(unmarshal func(interface{}) error) er
 
 	// Check for users putting URLs in target groups.
 	if len(c.RelabelConfigs) == 0 {
-		for _, tg := range c.ServiceDiscoveryConfig.StaticConfigs {
-			for _, t := range tg.Targets {
-				if err := CheckTargetAddress(t[model.AddressLabel]); err != nil {
-					return err
-				}
-			}
+		// TODO: can this either be enforced always or never?
```

Comment:
Always.

Reply:
Just to confirm... It's okay to push that check into the targetgroup.Group UnmarshalYAML function, so it's always applied regardless of if a RelabelConfig exists?

Reply:
No, relabel must exist. But if no relabel exist, it can fail on any SD's.

Reply:
Okay. Then I need to leave it as it is. I'll remove the TODO.

Reply:
This is because this is **valid**
```
scrape_configs:
  - job_name: 'blackbox'
    metrics_path: /probe
    params:
      module: [http_2xx]  # Look for a HTTP 200 response.
    static_configs:
      - targets:
        - http://prometheus.io    # Target to probe with http.
        - https://prometheus.io   # Target to probe with https.
        - http://example.com:8080 # Target to probe with http on port 8080.
    relabel_configs:
      - source_labels: [__address__]
        target_label: __param_target
      - source_labels: [__param_target]
        target_label: instance
      - target_label: __address__
        replacement: 127.0.0.1:9115  # The blackbox exporter's real hostname:port.
```

But this is **invalid**
```
scrape_configs:
  - job_name: 'blackbox'
    metrics_path: /probe
    params:
      module: [http_2xx]  # Look for a HTTP 200 response.
    static_configs:
      - targets:
        - http://prometheus.io    # Target to probe with http.
        - https://prometheus.io   # Target to probe with https.
        - http://example.com:8080 # Target to probe with http on port 8080.
```

Reply:
Yeah, I get it... My reason for asking was because the check didn't go deep into the RelabelConfigs to see if they replace the `__address__` label. I was thinking it might not be that important if relabeling something else was sufficient to bypass the check. People creating invalid targets is probably a big enough issue that it's worth keeping.